service/src/test/java/com/epam/rft/atsy/service/impl/PositionServiceImplTest.java

```diff
@@ -1,69 +1,186 @@
 package com.epam.rft.atsy.service.impl;
 
-/**
- * Unit tests for {@link PositionServiceImpl}.
- */
-/*public class PositionServiceImplTest {
-    @InjectMocks
-    private PositionServiceImpl underTest;
-    @Mock
-    private PositionDAO dao;
+import com.epam.rft.atsy.persistence.entities.PositionEntity;
+import com.epam.rft.atsy.persistence.repositories.PositionRepository;
+import com.epam.rft.atsy.service.domain.PositionDTO;
+import com.epam.rft.atsy.service.exception.DuplicateRecordException;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Matchers;
+import org.hibernate.exception.ConstraintViolationException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.TypeToken;
+import org.springframework.dao.DataIntegrityViolationException;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.times;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PositionServiceImplTest {
     @Mock
     private ModelMapper modelMapper;
 
-    @BeforeMethod
-    public void init() {
-        MockitoAnnotations.initMocks(this);
+    @Mock
+    private PositionRepository positionRepository;
+
+    @InjectMocks
+    private PositionServiceImpl positionService;
+
+    private static final Type POSITIONDTO_LIST_TYPE;
+
+    private static final Long DEVELOPER_ID = 1L;
+    private static final String DEVELOPER_NAME = "Developer";
+
+    private static final PositionEntity DEVELOPER_ENTITY;
+    private static final PositionDTO DEVELOPER_DTO;
+
+
+    private static final List<PositionEntity> EMPTY_POSITION_ENTITY_LIST = Collections.emptyList();
+    private static final Collection<PositionDTO> EMPTY_POSITION_DTO_LIST = Collections.emptyList();
+
+    private static final List<PositionEntity> EXPECTED_POSITION_ENTITY_LIST;
+    private static final Collection<PositionDTO> EXPECTED_POSITION_DTO_LIST;
+
+    static {
```

Comment:
The should be done in a method annotated by @BeforeClass

See here why: http://stackoverflow.com/questions/15493189/beforeclass-vs-static